Services/UI/OJS.Services.Ui.Business/Implementations/UsersBusinessService.cs

```diff
@@ -58,5 +59,26 @@ public async Task<bool> IsLoggedInUserAdmin(ClaimsPrincipal userPrincipal)
 
         public async Task AddOrUpdateUser(UserProfile user)
             => await this.usersProfileData.AddOrUpdate<UserProfileServiceModel>(user);
+
+        public async Task<UserAuthInfoServiceModel> GetAuthInfo()
+        {
+            var currentUser = this.userProvider.GetCurrentUser();
+
+            if (currentUser == null)
+            {
+                throw new BusinessServiceException("No user is currently in logged in session.");
+            }
```

Comment:
I would not throw exception, just return `null`. If the result is null we know on the client that the user is not logged in.